app/my_jss/accounts/Users.js

```diff
@@ -0,0 +1,79 @@
+var jsonUsers = JSON.parse(usersJSON);
+
+var userData = new DataController();
+userData.setUrlLocation(usersJSON);
+userData.setSaveName("users");
+
+function Users(){
+
+ this.allUsers = this.getAllUsers();
+}
+
+//set user state ////
+
+Users.prototype.setUserState = function(email,state){
+
+  var localUsers = this.allUsers;
+
+  for(var i=0; i < localUsers.length; i++){
+
+      if(localUsers.email === email)
+      {
+         localUsers.setLoginState(state);
+      }
+    }
+};
+// returns boolean
+Users.prototype.checkedByType = function(type,objectKey){
+
+
+  var localUsers = this.allUsers;
+   for(var i=0; i < localUsers.length; i++){
+
+      var user = localUsers[i];
+
+     if(user[type] === objectKey)
+      return true;
+   }
+ return false;
+};
+
+// returns array of all users//
+Users.prototype.getAllUsers = function () {
+
+  var thisOfUsers = this;
+
+  var mapFunction = function(val){
+
+      return  thisOfUsers.createUser(val.name, val.email, val.password);
```

Comment:
I used because i have a AddUser function to, and i didn't wanted to 
multiply code. Should i change?

On 24.03.2016 22:06, Cazaciuc Gabriel wrote:
>
> In app/my_jss/accounts/Users.js 
> <https://github.com/SOFTVISION-University/cluj-pre-js-2016/pull/94#discussion_r57379362>:
>
> > +      var user = localUsers[i];
> > +
> > +     if(user[type] === objectKey)
> > +      return true;
> > +   }
> > + return false;
> > +};
> > +
> > +// returns array of all users//
> > +Users.prototype.getAllUsers = function () {
> > +
> > +  var thisOfUsers = this;
> > +
> > +  var mapFunction = function(val){
> > +
> > +      return  thisOfUsers.createUser(val.name, val.email, val.password);
>
> Very good. createUser method i find it unnecesary because it doesn't 
> provide any value added. just do a return new User(...). Same for Song
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub 
> <https://github.com/SOFTVISION-University/cluj-pre-js-2016/pull/94/files/f3f0b4b846156df76bbbc044d5e2dadcf0ad901d#r57379362>
>